Jarvis.Framework.Shared/ReadModel/Atomic/ILiveAtomicReadModelProcessorEnhanced.cs

```diff
@@ -0,0 +1,83 @@
+﻿using MongoDB.Driver.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Jarvis.Framework.Shared.ReadModel.Atomic
+{
+	/// <summary>
+	/// Processor that is capable of live projection of atomic
+	/// readmodels, it will allow for historical projection when
+	/// you have a readmodel and you want to project up to a certain
+	/// point in time.
+	/// </summary>
+	public interface ILiveAtomicReadModelProcessorEnhanced
+	{
+		/// <summary>
+		/// Project a single aggregate into multiple readmodels, useful when we need
+		/// to project an aggregate that has multiple readmodels. #45
+		/// </summary>
+		/// <param name="types"></param>
+		/// <param name="id">If of the aggregate</param>
+		/// <param name="versionUpTo">Version of aggregate to project, it is NOT the global
+		/// position, but version of aggregate.
+		/// </param>
+		/// <returns></returns>
+		Task<MultipleReadmodelProjectionResult> ProcessAsync(IEnumerable<Type> types, String id, Int64 versionUpTo);

```

Comment:
non mi è molto chiaro il razionale del richiedere "up to version" non è detto che siano "allineate" o che abbiamo una qualche forma di relazione tra loro... in micro aggregate design non è proprio prevedibile quando uno degli stream cambia versione (o quando)... secondo me ha senso solo per date e position globale

Reply:
Io sto proiettando un unico aggregato, non più di un aggregato, solamente che ho più readmodel per lo stesso, ergo ho gli stessi X eventi che vengon dispatchati su più readmodel perché ho suddiviso i readmodel in più classi distinte.